### PR TITLE
Implement summary and metrics metadata fields

### PR DESCRIPTION
**Pull request by rsevilla87**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [x] Optimization
- [ ] Documentation Update

## Description

These changes are required to support indexing a different subset of metadata fields the indexed metrics different than jobSummary

## Related Tickets & Documents

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by vishnuchalla** [comment]
> These changes are required to support indexing a different subset of metadata fields the indexed metrics different than jobSummary

Shouldn't these different set of fields go into `jobSummary` as well? So that we will be able to fetch all the data without any miss when filters are applied in grafana.

File: pkg/util/metrics/metrics.go [hunk 1]
```diff
@@ -40,14 +38,15 @@ func ProcessMetricsScraperConfig(scraperConfig ScraperConfig) Scraper {
...
 		if err != nil {
 			log.Fatalf("Error reading provided user metadata: %v", err)
 		}
 	}
-	// Combine rawMetadata with user's provided metadata
-	for k, v := range scraperConfig.RawMetadata {
-		metadata[k] = v
+	// Combine users provided metadata with metrics and summary metadata
+	for k, v := range userMetadata {
+		scraperConfig.SummaryMetadata[k] = v
```
[the remark this answers is not in the document]
Comment (rsevilla87) [reply pkg/util/metrics/metrics.go:48]: The intent of having SummaryMetadata and MetricsMetadata is to store different information in them, the users metadata is common in both.

TLTR;
the content SummaryMetadata will be appended to the `jobSummary` doc and MetricsMetadata will be appended to the remaining docs (metrics, alerts, measurements...).

[the remark this answers is not in the document]
Comment (vishnuchalla) [reply pkg/util/metrics/metrics.go:48]: Understood. Can we also make sure that every field in `metricMetadata` is also in the `jobSummary` ensuring that its a subset of the summary.

[the remark this answers is not in the document]
Comment (vishnuchalla) [reply pkg/util/metrics/metrics.go:48]: I still feel like having separate data structure to hold summary metadata is not required while we only tend to add some additional user/metric metadata to the existing `jobSummary`.

**Comment by rsevilla87** [comment]
> > These changes are required to support indexing a different subset of metadata fields the indexed metrics different than jobSummary
> 
> Shouldn't these different set of fields go into `jobSummary` as well? So that we will be able to fetch all the data without any miss when filters are applied in grafana.

Good point, IMO, doing that it's something that should be handled in the wrapper implementation, as is at the moment it doesn't harm anything.

**Review by vishnuchalla — APPROVED** [review]
lgtm

**Pull request merged**